### ISSUE TITLE
SIGTERM is not handled too well

### ISSUE THREAD

**Issue by electrofloat** [authelia/authelia#4963]
### Version

v4.37.5

### Deployment Method

Bare-metal

### Reverse Proxy

NGINX

### Description

It seems that Authelia is stuck when it is sent the SIGTERM signal.

I had to modify the systemd service file to have this in it:
`TimeoutStopSec = 5`
so that it gets a SIGKILL after 5 seconds and not the default 90 seconds (which is the value of DefaultTimeoutStopSec on my system)

### Reproduction

kill -SIGTERM <authelia's pid>

### Expectations

Authelia shuts down gracefully on SIGTERM.

### Logs (Authelia)

```shell
Feb 19 14:56:56 localhost authelia-linux-amd64[36953]: time="2023-02-19T15:04:56+01:00" level=info msg="Authelia v4.37.5 is starting"
Feb 19 14:56:56 localhost authelia-linux-amd64[36953]: time="2023-02-19T15:04:56+01:00" level=info msg="Log severity set to debug"
Feb 19 14:58:39 localhost systemd[1]: authelia.service: State 'stop-sigterm' timed out. Killing.
Feb 19 14:58:39 localhost systemd[1]: authelia.service: Killing process 33794 (authelia-linux-) with signal SIGKILL.
```

### Pre-Submission Checklist

- [X] I agree to follow the [Code of Conduct](http://www.authelia.com/code-of-conduct)
- [X] This is a bug report and not a support request
- [X] I have read the security policy and this bug report is not a security issue or security related issue
- [X] I have either included the complete configuration file or I am sure it's unrelated to the configuration
- [X] I have provided all of the required information in full with the only alteration being reasonable sanitization in accordance with the [Troubleshooting Sanitization](https://www.authelia.com/r/sanitize) reference guide
- [X] I have checked for related proxy or application logs and included them if available
- [X] I have checked for related issues and checked the documentation

**Comment by james-d-elliott** [comment]
This issue is not complete, there are not a full set of logs.

**Comment by electrofloat** [comment]
What full set  of logs do you need, when authelia does not log anything when it gets the SIGTERM signal?

**Comment by james-d-elliott** [comment]
Process start -> problem occurring.

**Comment by electrofloat** [comment]
Is this better? I don't understand how this makes the issue more clear. Authelia is run by systemd. When systemd tries to stop it, it sends - as with any other service - the SIGTERM signal first. That is what makes authelia just "stuck" and not responding to anything (and also not making any new logs) so it has to kill it with SIGKILL.

It can be reproduced anytime as I described in the reproduction section without the logs.

**Comment by james-d-elliott** [comment]
Yes. It makes it more clear because it shows things that are meant to be happening not happening. It could have been included right when you read the issue template as it asked you to.

**Comment by james-d-elliott** [comment]
I believe I've fixed this, but I'm not 100% certain of the underlying cause in 4.37.5 as the fix was in relation to a for loop and go routines (virtual threads) which was introduced in a refactoring. I suspect the refactoring fixed the underlying cause of this issue and introduced another. In any case fix will be available in master soon and 4.38.0 when released.
